canonicalwebteam/http/__init__.py

```diff
@@ -100,24 +101,31 @@ class UncachedSession(BaseSession, requests.Session):
     pass
 
 
-class CachedSession(BaseSession, requests_cache.CachedSession):
+def create_cached_session(
```

Comment:
as discussed we should be consistant, or we use classes, or we use factories for both of the sessions. I think that in this case I'd rather use classes.

Reply:
I'd like to be included in that discussion. I advised @b-m-f to use a function (a "factory function") here, as there doesn't seem to be a neat way to subclass both our `BaseSession` (extension of `requests.Session`) and `CacheControl` together. If there is a neat way to create that class that doesn't lead to an object with a rather confusing type, then I'm happy to go with that.

But if the best way is, as I suspect, to simply pass our base session to the `CacheControl` class on initiation, then I think this should remain a function as that's the simplest construct. Creating a class for the sake of it is unnecessary complexity.

Reply:
I have updated this. Using the adapter seems to work neatly. I have outlined the problem of overwriting the TimeoutAdapter below. 
If needed I could dig deeper to see if chaining is possible somehow.